apps/backoffice/src/hooks/use-fetch.ts

```diff
@@ -0,0 +1,183 @@
+type ClientOperations = typeof clientWithBearerToken;
+
+/** Extract operation request parameters inferred by client operationId */
+type ExtractRequestParams<T extends keyof ClientOperations> = Parameters<
+  ClientOperations[T]
+>[0];
+
+/**
+ * Custom hook for client fetch API.
+ *
+ * This hook use `@pagopa/openapi-codegen-ts` autogenerated client, therefore makes available all and only APIs specified in the project root opeanapi.yaml
+ * @param operationId openapi operationId
+ * @param requestParams request parameters _(as specified in openapi)_
+ * @param responseCodec io-ts expected response codec
+ * @param redirectOnError if true, in case of fetch error, perform an auto redirect to an error page
+ * @param notifyOnError if true, in case of fetch error, show a notification
+ * @returns `data` fetch result of _responseCodec_ type
+ * @returns `error` in case of request error, decode failure or exceptions
+ * @returns `loading` as boolean value for request pending state
+ */
+const useFetch = <RC, T extends keyof ClientOperations>(
+  operationId: T,
+  requestParams: ExtractRequestParams<T>,
+  responseCodec: t.Type<RC>
+  // redirectOnError: boolean = false, // todo: will be implemented soon
+  // notifyOnError: boolean = false // todo: will be implemented soon
+) => {
+  const [data, setData] = useState<RC>();
+  const [error, setError] = useState<UseFetchError>();
+  const [loading, setLoading] = useState(true);
+  const { push } = useRouter();
+
+  /**
+   * SetStateAction\<UseFetchError>
+   * @param status
+   * @param message
+   * @returns
+   */
+  const setUseFetchError = (
+    kind: UseFetchErrorStatusType,
+    message: string,
+    status?: number
+  ) =>
+    setError({
+      kind,
+      message,
+      status
+    });
+
+  const fetchData = async (requestParams: any) => {
+    try {
+      const result = await clientWithBearerToken[operationId](requestParams);
+
+      if (result._tag === "Right") {
+        // Get client http response
+        const response = result.right;
+
+        // Check 401 Unauthorized
+        if (response.status === 401) {
+          push("/logout"); // todo: configure correct route path
+          return;
+        }
+
+        /** Set a response class type based on http response status code */
+        const httpResponseClassType = manageHttpResponseStatusCode(
+          response.status
+        );
+
+        switch (httpResponseClassType) {
+          case "successful":
+            // 2XX successful response status code
+            pipe(
+              responseCodec.decode(response.value),
+              E.fold(
+                e => setUseFetchError("validationError", readableReport(e)),
+                setData
+              )
+            );
+            break;
+          case "clientError":
+          case "serverError":
+            setUseFetchError(
+              "httpError",
+              httpResponseClassType,
+              response.status
+            );
+          default:
```

Comment:
**suggestion (non-blocking):** remove "useless" default.

If a default block does not have an implementation, simply omit it.
```suggestion
```

Reply:
925cfa7